petevdp/securedrop
# add confirmation flash when submission preferences are saved

## Status

Work in progress

## Description of Changes

Fixes #5009.

Changes proposed in this pull request:

## Testing

1. Run a dev instance
2. Navigate to /admin/config
3. Click the Update Submission Preferences button

outcome:
The page should refresh and a flashed message should be displayed, indicating that the preferences were successfully updated.

I haven't yet included any automated tests for this feature, but if that would be desirable I can definitely try to add those.


## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

## Checklist

### If you made changes to the server application code:

- [] Linting (`make lint`) and tests (`make test`) pass in the development container
Currently running tests, if there's an issue I'll post an update.
Instructions are somewhat unclear for linting via docker. The given `make lint` tries to lint from the host machine. Is there a preferred way to do this?

### If you made changes to documentation:
*Are there any changes to the documentation that might be required!
Unsure

### Improvements
* Currently the user is notified when the configuration is saved, regardless of whether or not they actually changed anything. Is this undesirable? If not, that should be simple enough to change.
